packages/vite/src/node/optimizer/index.ts

```diff
@@ -237,12 +238,29 @@ export async function optimizeDeps(
   const idToExports: Record<string, ExportsData> = {}
   const flatIdToExports: Record<string, ExportsData> = {}
 
+  const { plugins = [], ...esbuildOptions } =
+    config.optimizeDeps?.esbuildOptions ?? {}
+
   await init
   for (const id in deps) {
     const flatId = flattenId(id)
-    flatIdDeps[flatId] = deps[id]
-    const entryContent = fs.readFileSync(deps[id], 'utf-8')
-    const exportsData = parse(entryContent) as ExportsData
+    const filePath = (flatIdDeps[flatId] = deps[id])
+    const entryContent = fs.readFileSync(filePath, 'utf-8')
+    let exportsData: ExportsData
+    try {
+      exportsData = parse(entryContent) as ExportsData
+    } catch {
+      debug(
+        `Unable to parse dependency: ${id}. Trying again with a JSX transform.`
+      )
+      const transformed = await transformWithEsbuild(entryContent, filePath, {
+        loader: 'jsx'
+      })
+      // Ensure that optimization won't fail.
+      esbuildOptions.loader ??= {}
```

Comment:
We probably want to keep the `loader` mutation local to this function, so we should use spread syntax instead.

Reply:
Great catch, done.